zproject/backends.py

```diff
@@ -208,3 +215,7 @@ class DevAuthBackend(ZulipAuthMixin):
 
     def authenticate(self, username, return_data=None):
         return common_get_active_user_by_email(username, return_data=return_data)
+
+class GitHubBackend(ZulipAuthMixin, GithubOAuth2):
+    def complete(self, *args, **kwargs):
+        super(GitHubBackend, self).complete(*args, **kwargs)
```

Comment:
You'll probably need to add code similar to `common_get_active_user_by_email` for not allowing deactivated users/users in deactivated realms to login.